internal/components/generic_parser.go

```diff
@@ -67,15 +69,38 @@ func (g *GenericParser[T]) GetReadinessProbe(logger logr.Logger, config interfac
 	return g.readinessGen(logger, parsed)
 }
 
-func (g *GenericParser[T]) GetRBACRules(logger logr.Logger, config interface{}) ([]rbacv1.PolicyRule, error) {
-	if g.rbacGen == nil {
+func (g *GenericParser[T]) GetClusterRoleRules(logger logr.Logger, config interface{}) ([]rbacv1.PolicyRule, error) {
+	if g.clusterRoleRulesGen == nil {
 		return nil, nil
 	}
 	var parsed T
 	if err := mapstructure.Decode(config, &parsed); err != nil {
 		return nil, err
 	}
```

Comment:
this is other logic that would probably be good to abstract...

Reply:
Yes